wolfSSL/src/chapter02.md

```diff
@@ -536,10 +700,180 @@ Enables Online Certificate Status Protocol (OCSP) support.
+- WOLFSSL_HARDEN_TLS_ALLOW_OLD_TLS
+- WOLFSSL_HARDEN_TLS_NO_SCR_CHECK
+- WOLFSSL_HARDEN_TLS_NO_PKEY_CHECK
+- WOLFSSL_HARDEN_TLS_ALLOW_ALL_CIPHERSUITES
+
+#### WOLFSSL_ASIO
+
+OpenSSL compatibility specific macro.
+
+#### WOLFSSL_QT
+
+OpenSSL compatibility specific. Enable DH Extra for QT, OpenSSL all, OpenSSH, and static ephemeral.
+
+#### WOLFSSL_HAPROXY
+
+OpenSSL compatibility specific macro.
+
+#### WOLFSSL_ASN_TEMPLATE
+
+Use newer ASN template asn.c code (default). Daul algo certificate features. Dual alg cert support requires the ASN.1 template feature.
+
+#### WOLFSSL_ASYNC_IO
+
+Used in async cleanup.
+
+#### WOLFSSL_ATMEL
+
+Enables ASF hooks seeding random data using the `atmel_get_random_number` function.
+
+#### WOLFSSL_CMAC
+
+Additional CMAC algorithm enable. Note: requires `WOLFSSL_AES_DIRECT`.
+
+#### WOLFSSL_ESPIDF_ERROR_PAUSE
+
+Enable to pause in a loop rather than exit.
+
 #### TEST_IPV6
 
 Turns on testing of IPv6 in the test applications. wolfSSL proper is IP neutral, but the testing applications use IPv4 by default.
 
+#### TEST_NONBLOCK_CERTS
+
+Async test --enable-ocsp CFLAGS implement `TEST_NONBLOCK_CERTS`.
+
+#### TEST_OPENSSL_COEXIST
+
+Use when enabling the build option: `./configure --enable-opensslcoexist`.
+
+#### TEST_PK_PRIVKEY
+
+Loads and test PK private key.
+
+#### TEST_BUFFER_SIZE
+
+Can allocate buffer size for benchmarking.
+
+#### FORCE_BUFFER_TEST
+
+"Forces" buffer test in benchmark.
+
+#### WOLFSSL_FORCE_MALLOC_FAIL_TEST
+
+Define for internal testing to induce random malloc failures.
+
+#### WOLFSSL_POST_HANDSHAKE_AUTH
+
+TLS extension, Used for post-handshake authentification.
+
+#### WOLFSSL_PSK_MULTI_ID_PER_CS
+
+With TLS 1.3 PSK, when `WOLFSSL_PSK_MULTI_ID_PER_CS` is defined, multiple IDs for a cipher suite can be handled.
+
+#### WOLFSSL_PUBLIC_ASN
+
+This is needed to use ProcessPeerCert callback.
+
+#### WOLFSSL_QUIC
+
+Only "reads" from data provided by the application via wolfSSL_provide_quic_data(). Then, transfer from there into the inputBuffer. `WOLFSSL_QUIC` is incompatible with `WOLFSSL_CALLBACKS`.
+
+#### WOLFSSL_QUIC_H
+
+Define for wolSSL quic API.
+
+#### WOLFSSL_QUIC_MAX_RECORD_CAPACITY
+
+Defines max quic capacity as 1024*1024 -- 1 MB.
+
+#### WOLFSSL_RENESAS_FSPSM_TLS
+
+Not yet supported TLS related capabilities.
+
+#### WOLFSSL_RENESAS_TSIP_TLS
+
+This is for disabling only the TSIP TLS-linked common key encryption method.
+
+#### WOLFSSL_SM2
+
+Define to use SM ciphers.
+
+#### WOLFSSL_SM3
+
+Define to use SM ciphers.
+
+#### WOLFSSL_SM4
+
+Define to use SM ciphers.
+
+#### WOLFSSL_SM4_CBC
+
+SM setting for SM4 CBC.
+
+#### WOLFSSL_SM4_CCM
+
+SM settings for SM4 CCM.
+
+#### WOLFSSL_SM4_GCM
+
+SM settings for SM4 GCM.
+
+#### WOLFSSL_SNIFFER_CHAIN_INPUT
+
+Used in conjunction with build option `./configure --enable-sniffer`.
+
+#### XSLEEP_MS
+
+Can be called to make the session wait for a specified number of milliseconds till it expires.
```

Comment:
A macro for testing only that allows defining a custom delay. See wolfssl/test.h.